templates/quickstart-hashicorp-vault.template

```diff
@@ -312,364 +578,397 @@ Resources:
           - "-"
           - - Ref: AWS::StackName
-            - Vault2
-  Vault2RecoveryAlarm:
-    Type: AWS::CloudWatch::Alarm
+            - VaultClientSecurityGroup
+  VaultClientInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
     Properties:
-      AlarmDescription: EC2 Autorecovery for Vault2 node. Autorecover if we fail EC2
-        status checks for 5 minutes.
-      Namespace: AWS/EC2
-      MetricName: StatusCheckFailed_System
-      Statistic: Minimum
-      Period: 60
-      EvaluationPeriods: 5
-      ComparisonOperator: GreaterThanThreshold
-      Threshold: 0
-      AlarmActions:
-      - Ref: VaultSNSTopic
-      - Fn::Join:
-        - ''
-        - - 'arn:aws:automate:'
-          - Ref: AWS::Region
-          - ":ec2:recover"
-      Dimensions:
-      - Name: InstanceId
-        Value:
-          Ref: Vault2
+      Path: "/"
+      Roles:
+      - Ref: VaultClientRole
+  VaultClientRole:
+    Type: AWS::IAM::Role
+    Properties:
+      # This example allows for an EC2 instance to assume this role. This should be customized for where the Role is being used. Eg: Lambda/Codebuild/EC2 etc.
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - ec2.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Policies:
+      - PolicyName: root
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          # TODO: Restrict to the Lambda functions LeaderElection/ClusterBootstrapping
+          - Effect: Allow
+            Action:
+            - s3:*
+            Resource: "*"
+          # TODO: Adjust this to be a distinct policy to reference this role. (Circular dependency)
+          #- Effect: Allow
+          #  Action:
+          #  - sts:AssumeRole
+          #  Resource: !GetAtt "VaultClientRole.Arn"
+          - Effect: Allow
+            Action:
+            - ec2:DescribeInstances
+            - iam:GetInstanceProfile
+            - iam:GetUser
+            - iam:GetRole
+            Resource: "*"
+          - Effect: Allow
+            Action:
+            - autoscaling:DescribeAutoScalingGroups
+            Resource: "*"
 Outputs:
-  VaultNode1PrivateIp:
-    Description: 'Private IP of Vault Node #1'
-    Value:
-      Fn::GetAtt:
-      - Vault1
-      - PrivateIp
-  VaultNode2PrivateIp:
-    Description: 'Private IP of Vault Node #2'
-    Value:
-      Fn::GetAtt:
-      - Vault2
-      - PrivateIp
+  VaultSecret:
+    Value: !Ref "VaultSecret"
+    Description: The AWS Secrets Manager Secret containing the ROOT TOKEN and Recovery Secret for Hashicorp Vault.
+  VaultClientIAMRoleArn:
+    Value: !GetAtt "VaultClientRole.Arn"
+    Description: The ARN of the AWS IAM role linked to Hashicorp Vault.
+  VaultClientIAMRoleName:
+    Value: !Ref "VaultClientRole"
+    Description: The name of the AWS IAM role linked to Hashicorp Vault.
+  # TODO: Make this configurable
```

Comment:
Leave it not configurable. That's ok to post it out the outputs here.

Reply:
I believe I have actually made it configurable will confirm and remove the comment.